pkg/apis/pipeline/v1alpha1/verificationpolicy_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +genclient
+// +genclient:noStatus
+// +genreconciler:krshapedlogic=false
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerificationPolicy defines the rules to verify Tekon resources.
+// Tekton resources.
+// +k8s:openapi-gen=true
+type VerificationPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec holds the desired state of the VerificationPolicy.
+	Spec VerificationPolicySpec `json:"spec"`
+}
+
+// VerificationPolicyList contains a list of VerificationPolicy
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VerificationPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VerificationPolicy `json:"items"`
+}
+
+// GetGroupVersionKind implements kmeta.OwnerRefable.
+func (*VerificationPolicy) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("VerificationPolicy")
+}
+
+// VerificationPolicySpec defines the patterns and authorities.
+type VerificationPolicySpec struct {
+	// Resources defines the patterns of resources sources that should be subject to this policy.
+	// For example, we may want to apply this Policy from a certain github repo.
+	// Then the ResourcesPattern should be valid regex expression. E.g. If using gitresolver, and we want to config keys from a certain git repo.
```

Comment:
Grammar nit: `should be a valid regex`

(IMO: "regex expression" is redundant, since "regex" stands for "regular expression" -- but reasonable people may disagree with me.)

Reply:
👍